File: internal/content/testdata/bench/go/nc-meeting-rooms-ii.go
func min_meeting_rooms(intervals [][]int) int {
	if len(intervals) == 0 {
		return 0
	}
	n := len(intervals)
	starts := make([]int, n)
	ends := make([]int, n)
	for i, iv := range intervals {
		starts[i] = iv[0]
		ends[i] = iv[1]
	}
	// sort both (simple insertion sort)
	for i := 1; i < n; i++ {
		for j := i; j > 0 && starts[j] < starts[j-1]; j-- {
			starts[j], starts[j-1] = starts[j-1], starts[j]
		}
	}
	for i := 1; i < n; i++ {
		for j := i; j > 0 && ends[j] < ends[j-1]; j-- {
			ends[j], ends[j-1] = ends[j-1], ends[j]
		}
	}
	rooms, best := 0, 0
	s, e := 0, 0
	for s < n {
		if starts[s] < ends[e] {
			rooms++
			s++
			if rooms > best {
				best = rooms
			}
		} else {
			rooms--
			e++
		}
	}
	return best
}

File: internal/content/testdata/bench/rust/nc-meeting-rooms-ii.rs
fn min_meeting_rooms(intervals: Vec<Vec<i64>>) -> i64 {
    if intervals.is_empty() {
        return 0;
    }
    let mut starts: Vec<i64> = intervals.iter().map(|i| i[0]).collect();
    let mut ends: Vec<i64> = intervals.iter().map(|i| i[1]).collect();
    starts.sort();
    ends.sort();
    let mut rooms: i64 = 0;
    let mut best: i64 = 0;
    let mut s = 0usize;
    let mut e = 0usize;
    while s < starts.len() {
        if starts[s] < ends[e] {
            rooms += 1;
            s += 1;
            if rooms > best {
                best = rooms;
            }
        } else {
            rooms -= 1;
            e += 1;
        }
    }
    best
}
